enumerables.rb

```diff
@@ -0,0 +1,190 @@
+        my_each do |item|
+          return false unless item.instance_of? arg
+        end
+      else
+        my_each do |item|
+          return false unless item == arg
+        end
+      end
+    elsif block_given?
+      my_each do |item|
+        eval = yield item
+        return false if eval == false or eval == nil?
+      end
+    elsif !block_given? && arg.nil?
+      my_each do |item|
+        return false if item.nil? or item == false
+      end
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if !arg.nil?
+      if arg.is_a? Regexp
+        my_each do |item|
+          return true if item.to_s.match(arg)
+        end
+      elsif arg.is_a? Class
+        my_each do |item|
+          return true if item.instance_of? arg
+        end
+      else
+        my_each do |item|
+          return true if item == arg
+        end
+      end
+    elsif block_given?
+      my_each do |item|
+        eval = yield item
+        return true if eval == true
+      end
+    elsif !block_given? && arg.nil?
+      my_each do |item|
+        return true if item.nil? == false or item != false
+      end
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    if !arg.nil?
+      if arg.is_a? Regexp
+        my_each do |item|
+          return false if item.to_s.match(arg)
+        end
+      elsif arg.is_a? Class
+        my_each do |item|
+          return false if item.instance_of? arg
+        end
+      else
+        my_each do |item|
+          return false if item == arg
+        end
+      end
+    elsif block_given?
+      my_each do |item|
+        eval = yield item
+        return false if eval == true
+      end
+    elsif !block_given? && arg.nil?
+      my_each do |item|
+        return false unless item.nil? or item == false
+      end
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    counter = 0
+    return length if arg.nil? and !block_given?
+
+    if block_given?
+      my_each do |item|
+        counter += 1 if yield item
+      end
+    else
+      my_each do |item|
+        counter += 1 if item == arg
+      end
+    end
+    counter
+  end
+
+  def my_map(&block)
+    mapped = []
+    exec = block if block_given?
+    if exec.is_a? Proc
+      my_each do |item|
+        mapped << exec.call(item)
+      end
+      return to_enum(:my_map) unless exec.is_a? Proc
+    end
+
+    mapped
+  end
+
+  def my_inject(arg = nil, symb = nil)
+    temp_self = if is_a? Range
+                  to_a
+                else
+                  self
+                end
+
+    memo = if arg.nil? or arg.is_a? Symbol
+             first
+           else
+             arg
+           end
+    symb = arg if arg.is_a? Symbol
+    unless symb.nil?
+      mymethod = symb.to_s
+      for item in 1...to_a.length
+
+        memo = eval "#{memo}#{mymethod}#{temp_self[item]}"
+      end
+    end
+    if block_given?
+      for item in 1...length
+        memo = yield memo, self[item]
+      end
+    end
+    memo
+  end
```

Comment:
`my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- combines all elements of enum by applying a binary operation, specified by a block
```ruby
[1, 2, 3].my_inject(1) { |memo, num| memo + num } #should return 7
(1..3).my_inject(1) { |memo, num| memo + num } #should return 7
%w[dog door rod blade].my_inject {|memo, word| memo.length > word.length ? memo : word}  #should return 'blade'
```
- when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
[1, 2, 3].my_inject(1, :+)  #should return 7
(1..3).my_inject(1, :+)  #should return 7
(5..10).my_inject(2, :*)  #should return 302400
```